### PR TITLE
[camerax] Prevent using unsupported concurrent camera use cases

### PR DESCRIPTION
**Pull request by camsim99**
Adds logic + documentation to prevent the usage of unsupported (by CameraX) concurrent camera use cases.

The CameraX plugin should only support the concurrent camera use cases supported by Camerax; see [their documentation](https://developer.android.com/media/camera/camerax/architecture#combine-use-cases) for more information. To avoid the usage of unsupported concurrent use cases, this PR changes the plugin so that it behaves according to the following:

* If the preview is paused (via `pausePreview`), concurrent video recording and image streaming (via `startVideoCapturing(cameraId, VideoCaptureOptions(streamCallback:...))`) is supported.
* If the preview is not paused
  * **and** the camera device is at least supported hardware [`LIMITED`](https://developer.android.com/reference/android/hardware/camera2/CameraMetadata#INFO_SUPPORTED_HARDWARE_LEVEL_LIMITED), then concurrent
    image capture and video recording is supported.
  * **and** the camera device is at least supported hardware [`LEVEL_3`](https://developer.android.com/reference/android/hardware/camera2/CameraMetadata#INFO_SUPPORTED_HARDWARE_LEVEL_3), then concurrent video recording and image streaming is supported, but if used, concurrent image capture alongside this is not supported.

Also changes `Surface` constants to match Dart style and small documentation nits I noticed.

Fixes https://github.com/flutter/flutter/issues/146395, fixes https://github.com/flutter/flutter/issues/144414.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera_android_camerax/test/camera2_camera_info_test.mocks.dart [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        returnValue: 0,
+      ) as int);
+
+  @override
+  int getSupportedHardwareLevel(int? identifier) => (super.noSuchMethod(
+        Invocation.method(
+          #getSupportedHardwareLevel,
+          [identifier],
+        ),
+        returnValue: 0,
```
Comment (reidbaker) [inline packages/camera/camera_android_camerax/test/camera2_camera_info_test.mocks.dart:76]: Consider a comment saying this is LIMITED. It coudl help in debugging later to know that the device might not have returned limited but we had an error. https://developer.android.com/reference/android/hardware/camera2/CameraMetadata#INFO_SUPPORTED_HARDWARE_LEVEL_LIMITED

**Review by reidbaker — APPROVED** [review]
I glossed over the code that appeared generated.

**Comment by camsim99** [comment]
Just realized I didn't check for LIMITED level devices to disable image capture...just wanted to note I need to add this check + test before landing this!

File: packages/camera/camera_android_camerax/test/camera2_camera_info_test.mocks.dart (hunk 1 above)
Comment (camsim99) [reply packages/camera/camera_android_camerax/test/camera2_camera_info_test.mocks.dart:76]: This is generated code :/ I think it's just a mock value

**Review by reidbaker — APPROVED** [review]

**Pull request merged**